src/eureka/S3_data_reduction/s3_reduce.py

```diff
@@ -232,12 +225,50 @@ def reduce(eventlabel, ecf_path=None, s2_meta=None, input_meta=None):
                 raise ValueError('NIRISS observations are currently '
                                  'unsupported!')
             elif meta.inst == 'wfc3':
+                if 'jwst-crds.stsci.edu' in os.environ['CRDS_SERVER_URL']:
+                    log.writelog('CRDS_SERVER_URL is set for JWST and not HST.'
+                                 ' Automatically adjusting it to hst.')
+                    crds.client.api.set_crds_server(
+                        'https://hst-crds.stsci.edu/')
+                    os.environ['CRDS_SERVER_URL'] = \
+                        'https://hst-crds.stsci.edu/'
+
+                # If a specific CRDS context is entered in the ECF, apply it.
+                # Otherwise, log and fix the default CRDS context to make sure
+                # it doesn't change between different segments.
+                if not hasattr(meta, 'pmap') or meta.pmap is None:
+                    # Get just the numerical value
+                    meta.pmap = crds.get_context_name('hst')[4:-5]
+                os.environ['CRDS_CONTEXT'] = f'hst_{meta.pmap}.pmap'
```

Comment:
By having `meta.pmap = hst_1234`, I think we can avoid repeating some code here and on Lines 254-257.  Thoughts?